backend/redis/health_redis.go

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2019 by PACE Telematics GmbH. All rights reserved.
+// Created at 2019/10/18 Charlotte Pröller
+
+package redis
+
+import (
+	"github.com/go-redis/redis"
+	"github.com/pace/bricks/maintenance/health/servicehealthcheck"
+	"time"
+)
+
+type redisHealthCheck struct {
+	State  *servicehealthcheck.ConnectionState
+	client *redis.Client
+}
+
+func (h *redisHealthCheck) InitHealthCheck() error {
+	h.client = Client()
+	h.State = servicehealthcheck.NewConnectionState()
+	return nil
+}
+
+// HealthCheck checks if the redis is healthy. If the last result is outdated, redis is checked for writeability and readability,
+// otherwise return the old result
+func (h *redisHealthCheck) HealthCheck() (bool, error) {
+	currTime := time.Now()
+	if currTime.Sub(h.State.LastCheck) > cfg.HealthMaxRequest {
+		h.State.SetHealthy(currTime)
```

Comment:
If there is a second call to `HealthCheck()` while the first one is being processed, the state is reported back as healthy, but in fact it should be the previous status. This is because this line sets the state to healthy without the check being processed. The second call will skip the test because of `cfg.HealthMaxRequest` and report it being healthy.